### PR TITLE
add icon pattern to fields

### PR DESCRIPTION
**Pull request by fenekku**
this adds an icon to the existing field and introduces `props` for label icons going forward:

`<MyComponent label={"Label text"} labelIcon={"<semantic-ui icon name>"} ... />`

### CODE REVIEW

File: invenio_app_rdm/theme/assets/js/react_invenio_deposit/components/ResourceTypeField.js [hunk 1]
```diff
@@ -8,45 +8,49 @@
...
+import { Form, Icon } from "semantic-ui-react";
 
 import { SelectField } from "../../react_invenio_forms";
 
+
 export class ResourceTypeField extends Component {
   renderResourceTypeField = (formikBag) => {
-    const resource_type_fieldpath = `${this.props.fieldPath}.type`;
-    const resource_subtype_fieldpath = `${this.props.fieldPath}.subtype`;
+    const typeFieldPath = `${this.props.fieldPath}.type`;
```
Comment (zzacharo) [inline invenio_app_rdm/theme/assets/js/react_invenio_deposit/components/ResourceTypeField.js:18]: Nice :)

File: invenio_app_rdm/theme/assets/js/react_invenio_deposit/components/ResourceTypeField.js [hunk 2]
```diff
@@ -8,45 +8,49 @@
...
+      formikBag.form.setFieldValue(subtypeFieldPath, "");
     };
 
-    const subtype_options = this.props.options.subtype.filter(
+    const subtypeOptions = this.props.options.subtype.filter(
       (e) => e["parent-value"] === resource_type
     );
     return (
       <div>
+        <label><Icon name={this.props.labelIcon} />{this.props.label}</label>
```
Comment (zzacharo) [inline invenio_app_rdm/theme/assets/js/react_invenio_deposit/components/ResourceTypeField.js:40]: @fenekku Here, it seems that the icon is a required prop but not defined as one in the propTypes section. Should we require the use of an icon, do you think? Is this coming directly from the vocabulary endpoint?

Comment (fenekku) [reply invenio_app_rdm/theme/assets/js/react_invenio_deposit/components/ResourceTypeField.js:40]: The icon shouldn't be required. I wasn't getting errors when `labelIcon` wasn't passed, but I now I see that the `Icon` component still renders which takes whitespace/looks strange. I will make it not appear if `labelIcon` is not passed to look better :).

**Comment by fenekku** [comment]
- only display Icon if an icon prop is passed.
- made label a non-required prop to make the component more flexible
- added `FieldLabel` as a reusable component in react-invenio-forms to take care of the label presentation logic

File: invenio_app_rdm/theme/assets/js/react_invenio_forms/core/FieldLabel.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    const {htmlFor, icon, label} = this.props;
+    return (
+      <label htmlFor={htmlFor}>{icon ? <Icon name={icon} /> : null}{label}</label>
+    )
+  }
+}
+
+FieldLabel.propTypes = {
+  htmlFor: PropTypes.string,
+  label: PropTypes.string,
```
Comment (zzacharo) [inline invenio_app_rdm/theme/assets/js/react_invenio_forms/core/FieldLabel.js:23]: Just mark the props that are required here, e.g `label`, `htmlFor`. Also a nice trick for optional props that you want to give a default value is this https://reactjs.org/docs/typechecking-with-proptypes.html#default-prop-values

Comment (fenekku) [reply invenio_app_rdm/theme/assets/js/react_invenio_forms/core/FieldLabel.js:23]: I thought `propTypes` was a nice way to define the interface for the component, but it should only be used for required props? So if a prop is not required, it should not be placed in the `propTypes`but rather only be placed in the `defaultProps`? Did I get this right?

What if there is no sensible default?: for icon there isn't a default - I guess `null` or `undefined` could be in the default, but kind of implied...

**Comment by fenekku** [comment]
Closed via #114

**Pull request closed without merging**